db/memtable.cc

```diff
@@ -842,6 +846,66 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
   return found_final_value;
 }
 
+bool MemTable::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
+                        ReadCallback* callback, bool* is_blob) {
+  // The sequence number is updated synchronously in version_set.h
+  if (IsEmpty()) {
+    // Avoiding recording stats for speed.
+    return false;
+  }
+  std::bitset<MultiGetContext::MAX_BATCH_SIZE> keys;
+  int idx = 0;
+  for (auto iter = range->begin(); iter != range->end(); ++iter) {
```

Comment:
Create a new MultiGetRange from the input range parameter. Then use the new range in this function, as well as pass it down to ```bloom_filter_->MayContain()```. It will allow you to simplify the rest of the function, as explained in the following comments.

Using a new range here also ensures that the original range remains intact, and can be used for subsequent stages of ```DB::MultiGet```.